### DISCUSSION TITLE
Improving WFDB

### DISCUSSION THREAD

**Discussion by briangow** [chorus-ai/chorus_waveform#88] [General]
As mentioned in https://github.com/chorus-ai/chorus_waveform/discussions/11#discussioncomment-9773185 , the waveform standardization working group recommended WFDB as the basis for a waveform format for CHoRUS. 

We would like to improve WFDB, both for CHoRUS and in general. We are currently planning improvements in these areas:

- Documentation (in particular around making it easier for new users to get up to speed on using WFDB)
- Python package performance

**Comment by briangow** [comment]
There has also been some discussion around updates to the annotation standard in WFDB:

- Point annotations - it would be nice to be able to have multiple annotations at a given point without the need for a separate file for each annotation. 
- Hierarchical structure - we could consider extending WFDB to allow it to use something like XML to create an annotation set. Since this could affect performance we need to keep that trade-off in mind. 

We haven't decided whether to make these updates, so please add to the discussion around this if you have a preference and include suggestions for how to do the updates, if applicable.

**Comment by clermontg** [comment]
A couple of initial thoughts:
It would be nice to accommodate irregularly sampled data. The DWC enumerations files are just this, as are some numerics. 
Easy recipes to store computed fields, within WFBD or not

**Comment by gariclifford** [comment]
It would be wonderful to have WFDB support (through a routine to write a flat file header) Apache Parquet, MAT 7.3+ / HDF5. This would cover most file formats modern data scientists use.
